src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -106,17 +105,23 @@ public CandlepinQuery<Pool> listByOwner(Owner owner) {
      * @return pools owned by the given Owner.
      */
     @Transactional
-    public CandlepinQuery<Pool> listByOwner(Owner owner, Date activeOn) {
-        DetachedCriteria criteria = DetachedCriteria.forClass(Pool.class)
-            .add(Restrictions.eq("owner", owner))
-            .add(Restrictions.eq("activeSubscription", Boolean.TRUE));
+    public List<Pool> listByOwner(Owner owner, Date activeOn) {
+        String jpql = "SELECT p FROM Pool p " +
+            "WHERE p.owner.id = :owner_id AND p.activeSubscription = true ";
+
+        if (activeOn != null) {
```

Comment:
When we're doing stuff like dynamically building JPQL strings, we should evaluate if it'd be better to use JPA criteria instead.

Reply:
Good point, this works better using JPA criteria. Updated